### DISCUSSION TITLE
from-json(string, fallback?) function not working as expected

### DISCUSSION THREAD

**Discussion by SavinayaShetty** [schibsted/jslt#220] [Q&A]
JSLT functions Document [(ref link)](https://github.com/schibsted/jslt/blob/master/functions.md#from-jsonstring-fallback---value) states following about from-json(string, fallback?) function-
> Parses the string as JSON and returns the result. If the string is null then the function will return null

EX: 1 - Working Fine
input:
```
{
   "displayText" : "Subscribe",
   "postback" : "{\"name\":\"John\",\"in_out\":\"in\"}"
}
```
template:
`from-json(.postback, null)`
output:
```
{
  "name" : "John",
  "in_out" : "in"
}
```
As per the template if the postback is not a valid json, it should return null. But in the following case, it is giving unusual output.
EX: 2 - Unusual output
input:
```
{
   "displayText" : "Subscribe",
   "postback" : "\"name\":\"John\",\"in_out\":\"in\""
}
```
template:
`from-json(.postback, null)`
output:
`"name"`

Is this a bug in 'from-json' function?

**Comment by larsga** [comment]
Yes, that's a bug. Thank you for reporting it so clearly.

Looks like when `from-json` is parsing the string it stops after `"name"` instead of reporting the parse error like it should.

**Reply by SavinayaShetty** [reply]
Yes. I checked the code, seems like the issue is from Jackson ObjectMapper itself, 
> JsonNode parsed = NodeUtils.mapper.readTree(json);

I'll read up on this and see what I can find out. Thanks.
